### DISCUSSION TITLE
Superformula as a path for a voxelfield?

### DISCUSSION THREAD

**Discussion by 5RIndustries** [leap71/PicoGK#20] [Q&A]
Hey team, I apologize in advance if this question makes no sense, but I am trying to understand the mindset behind, or that influences, a design using PicoGK.   Seems easy to create limits in voxelspace that create edges/corners, but I don't see where you are adding in traditional/external 3D geometry that creates them.  Specifically, how do you use or express superformula as a path for voxelfields?  Or are all the paths created with 3d  Euclidean distance from a centerpoint, and then the voxelfield created as radii from the path?  Or are you just adding voxfields together like blocks? - thanks for your help - Adrian.    

EDIT 1:  Sorry, I took a lot of the jibberish out and did some more research to help me understand my own question.

EDIT 2: Does PicoGK account for the overlap or compression/stretch around any potential curvature/geodesic in the path, or do you have to design the voxelfields in linear forms?

EDIT 3:  More specifically, can PicoGK support conformal transformation calculations of voxelfields?

**Comment by LinKayser** [comment]
Hi there — I think the best way to "wrap your head around" (sorry for the pun) curved objects is to have a look at the Shape Kernel documentation, which shows how to work with base shapes and warp them around a spine, use local coordinate systems. https://github.com/leap71/LEAP71_ShapeKernel

You will not stretch/transform the voxel field directly (this would create all kinds of aliasing artifacts, because of the discretized Euclidian nature of the field, similar to what you would see in geometric transformations of a 2D bitmap). Instead you will work on a higher level, with meshes or lattices, and then transform those. This is how we create things like the volutes of our rocket engines, or the pipes of our heat exchangers.

![0DEAC545-9E5A-4E20-8B8E-5EAD2C4AD041](https://github.com/leap71/PicoGK/assets/148448288/ff2d3a21-5d9b-427d-97f0-eea403fcf2d0)

![EE4231CF-9A36-4580-8D01-4E0EEE71D962](https://github.com/leap71/PicoGK/assets/148448288/b76e79cc-114c-4fd8-8f39-fb35fb3a9cf0)

Please let me know if you get stuck. This document here is probably the best starting point:

https://github.com/leap71/LEAP71_ShapeKernel/blob/main/Documentation/README-ReadingDetails.md
